app/serializers/device_serializer.rb

```diff
@@ -0,0 +1,7 @@
+class DeviceSerializer < ActiveModel::Serializer
+  attributes *%i(id location location_name payload created_at updated_at)
+
+  def location
```

Comment:
I see. I deleted this method. All data goes into payload, we don't lose any data.